data_structures/src/data_request.rs

```diff
@@ -204,47 +204,53 @@ impl DataRequestPool {
                 // We can notify the user that a data request from "my_claims" is available
                 // for reveal.
                 if dr_state.update_stage() {
```

Comment:
`update_stage` only returns true when the stage changes

Reply:
I think you can just remove the `if` and everything will work as expected

```suggestion
                dr_state.update_stage();
```